ext/bg/js/backend.js

```diff
@@ -968,6 +945,60 @@ class Backend {
 
     // Utilities
 
+    _getModifySettingObject(target) {
+        switch (target.scope) {
+            case 'profile':
+                if (!isObject(target.optionsContext)) { throw new Error('Invalid optionsContext'); }
+                return this.getOptions(target.optionsContext, true);
+            case 'global':
+                return this.getFullOptions(true);
+        }
+    }
```

Comment:
This is good for correctness, but looking at the object "signature" it feels like there could be a separate `apiModifySettingsGlobal` in addition to `apiModifySettingsProfile`. Alternatively classes that use `apiModifySettings` could wrap the function so that it automatically uses the `optionsContext`, `action`, `scope` and `source` that's needed. Don't have to do this now but if this starts to get used more somewhere it could be useful.

Also this could throw by default.

Reply:
> Also this could throw by default.

Forgot to do that. b2908abea1418da85521fc739ab4ed28f89ca9e6

> This is good for correctness, but looking at the object "signature" it feels like there could be a separate `apiModifySettingsGlobal` in addition to `apiModifySettingsProfile`.

I think the use of this will probably make a bit more sense once it's used on the settings page. Effectively there will be a `<input ... data-scope="global">` for some of the options. It could be done the way you mentioned as well, but this might be more simple. We'll see as it becomes used.